src/mondemand_statsmsg.erl

```diff
@@ -2010,6 +2082,51 @@ statsmsg_test_ () ->
                     lists:seq (0,1024)),
         ?assertEqual (true, R)
       end
+    },
+    { "many metrics",
+      fun() ->
+        C = [{<<"foo">>,<<"bar">>}],
+        MetricsCount = ?MD_MAX_METRICS + 1,
+        MIn = lists:map(fun(X) ->
+                          {
+                            gauge,
+                            list_to_binary(string:concat("baz", integer_to_list(X))),
+                            X
+                          } end,
+                        lists:seq(1, MetricsCount)),
+        S = new (<<"program_id">>, C, MIn, <<"known">>, 5, 5),
+        ?assertEqual (5, send_time (S)),
+        % check adding extra contexts
+        SwC = add_context (S, <<"blah">>, <<"boo">>),
+        ?assertEqual (<<"boo">>, context_value (SwC, <<"blah">>)),
+        SwCL = add_contexts (S, [{<<"blah">>,<<"boo">>},{<<"aa">>,<<"zz">>}]),
+        ?assertEqual (<<"boo">>, context_value (SwCL, <<"blah">>)),
+        ?assertEqual (<<"zz">>, context_value (SwCL, <<"aa">>)),
+        % test that to_lwes of long statsmsg  returns several events that are correct
+        [E1, E2] = to_lwes(S),
+        % test that the metric counts and first metric in events match expected values
+        %?debugFmt("~p~n", [E2#lwes_event.attrs]),
+        ?assert (lists:member({uint16, <<"num">>, ?MD_MAX_METRICS}, E1#lwes_event.attrs)),
```

Comment:
I think there are some defines you can use instead of the raw atoms for the lwes types.  Check lwes.hrl.